### ISSUE TITLE
Compatibility with gtk-layer-shell

### ISSUE THREAD

**Issue by Cu3PO42** [Cu3PO42/gtk-session-lock#4]
It appears that gtk-session-lock and gtk-layer-shell are currently mutually exclusive. Which ever is used first breaks the other one. This is likely a name clash of some symbol.

**Comment by Cu3PO42** [comment]
I have marked most symbols as hidden, this should resolve the above issue.

**Comment by Cu3PO42** [comment]
Marking symbols as private did indeed resolve the issue that one library always reports itself as unsupported, but leaves the following issue:

```
(com.github.Aylur.ags:105088): GLib-GObject-CRITICAL **: 00:10:34.133: ../gobject/gsignal.c:2088: type 'GtkWindow' is already overridden for signal id '47'

(com.github.Aylur.ags:105088): GLib-GObject-CRITICAL **: 00:10:34.133: ../gobject/gsignal.c:2088: type 'GtkWindow' is already overridden for signal id '46'
```

This is likely related to some hooks that both libraries install. It's currently unclear how to resolve that issue.

**Comment by kotontrion** [comment]
It is still required that gtk-layer-shell is initialized first, otherwise it will crash when using popups like tooltips.

This might be resolvable by subclassing Gtk.Window instead of overriding the realize and unmap functions of **every** Gtk.Window.

**Comment by Cu3PO42** [comment]
I resolved this slightly differently. Unfortunately hooking the `unmap` function like gtk-layer-shell does is the only way to run before the native unmap handler and it can only be used by one library at the time. A "clean" fix would require moving some abstractions into a seperate library that gtk-layer-shell and gtk-session-lock both use.

My somewhat more hacky, but much easier to fix is to require any users of the library to manually call a function before they hide a lock window. I'm okay with this because I can't think of a good reason to ever do this. If you hide a lock window, the compositor will just render a solid color (in practice: red) in its place. Destroying the window after you're done requires no workaround.

This is resolved in v0.2.
